datadog_checks_dev/datadog_checks/dev/tooling/manifest_validator/common/validator.py

```diff
@@ -0,0 +1,273 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import abc
+import os
+from typing import Dict
+
+import six
+
+from datadog_checks.dev.tooling.git import content_changed
+from datadog_checks.dev.tooling.utils import get_metadata_file, has_logs, is_metric_in_metadata_file, read_metadata_rows
+
+
+class ValidationResult(object):
+    def __init__(self):
+        self.failed = False
+        self.fixed = False
+        self.messages = {'success': [], 'warning': [], 'failure': [], 'info': []}
+
+    def __str__(self):
+        return '\n'.join(['\n'.join(messages) for messages in self.messages.values()])
+
+    def __repr__(self):
+        return str(self)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseManifestValidator(object):
+    def __init__(
+        self,
+        is_extras=False,
+        is_marketplace=False,
+        check_in_extras=True,
+        check_in_marketplace=True,
+        ctx=None,
+        version="v1",
+        skip_if_errors=False,
+    ):
+        self.result = ValidationResult()
+        self.is_extras = is_extras
+        self.is_marketplace = is_marketplace
+        self.check_in_extras = check_in_extras
+        self.check_in_markeplace = check_in_marketplace
+        self.ctx = ctx
+        self.version = version
+        self.skip_if_errors = skip_if_errors
+
+    def should_validate(self):
+        if not self.is_extras and not self.is_marketplace:
+            return True
+        if self.is_extras and self.check_in_extras:
+            return True
+        if self.is_marketplace and self.check_in_markeplace:
+            return True
+        return False
+
+    def validate(self, check_name, manifest, should_fix):
+        # type: (str, Dict, bool) -> None
+        """Validates the decoded manifest. Will perform inline changes if fix is true"""
+        raise NotImplementedError
+
+    def fail(self, error_message):
+        self.result.failed = True
+        self.result.messages['failure'].append(error_message)
+
+    def fix(self, problem, solution):
+        self.result.warning_msg = problem
+        self.result.success_msg = solution
+        self.result.fixed = True
+        self.result.failed = False
+
+    def __repr__(self):
+        return str(self.result)
+
+
+class MaintainerValidator(BaseManifestValidator):
+
+    MAINTAINER_PATH = {"v1": '/maintainer', "v2": '/author/support_email'}
+
+    def validate(self, check_name, decoded, fix):
+        if not self.should_validate():
+            return
+        correct_maintainer = 'help@datadoghq.com'
+
+        path = self.MAINTAINER_PATH[self.version]
+        maintainer = decoded.get_path(path)
+
+        if not maintainer.isascii():
+            self.fail(f'  `maintainer` contains non-ascii character: {maintainer}')
+            return
```

Comment:
I think this is something we'll want to check for marketplace/extras. Any reason we skip testing this validation on those integration types? (and instead just put the `correct_maintainer` behind a core only check?)

Reply:
Guessing the rationale was because we couldn't fix anything other than core maintainers.

We should probably have another validation later that does broader stroke ASCII validation.